src/vs/workbench/contrib/comments/browser/commentsView.ts

```diff
@@ -99,11 +105,29 @@ export class CommentsPanel extends FilterViewPane implements ICommentsView {
 
 		this._register(this.commentService.onDidSetAllCommentThreads(e => {
 			this.totalComments += e.commentThreads.length;
+
+			let unresolved = 0;
+			for (const thread of e.commentThreads) {
+				if (thread.state === CommentThreadState.Unresolved) {
+					unresolved++;
+				}
+			}
+			this.updateBadge(unresolved);
 		}));
 
 		this._register(this.commentService.onDidUpdateCommentThreads(e => {
 			this.totalComments += e.added.length;
 			this.totalComments -= e.removed.length;
+
+			let unresolved = 0;
+			for (const resource of this.commentsModel.resourceCommentThreads) {
+				for (const thread of resource.commentThreads) {
+					if (thread.threadState === CommentThreadState.Unresolved) {
```

Comment:
`this.commentsModel.resourceCommentThreads` has not yet been updated when `onDidUpdateCommentThread` fires, resulting in a low count of unresolved.

Reply:
Is there another event we could use?

Reply:
The problem was that we were listening to the same event twice and so didn't control the ordering. Moving everything into the same listener resolved this.